### DISCUSSION TITLE
moose about re-compile

### DISCUSSION THREAD

**Discussion by wftxd** [idaholab/moose#28897] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

Hello,

I have been learning MOOSE recently, but I encountered a problem. Every time I modify a header file or source file, I have to enter the following commands from the official website:
make clobberall
make -j 6
to achieve recompilation. However, I realize that this takes too much time. How can I improve this?

For example, if I only modify example.h, using 
make clobberall  
make -j 6 
will recompile all my files, which takes a lot of time. Is there a command that allows me to only recompile the example.h file?

I believe this will greatly increase my efficiency. Much appreciated for advice.

**Comment by GiudGiud** [comment]
hello

you should not need to `make clobberall`.
`make -j6` should be enough to recompile for most cases

Guillaume

**Reply by wftxd** [reply]
Do you mean that every time I modify a header file or source file, I have to enter make -j6 to recompile? Is there no way to perform this operation for just a single file?

**Reply by GiudGiud** [reply]
It s very fast. 

It should only recompile the one file (or one unity file) that got changed + any file that depends on that header

When you used make clobberall you were forcing it to recompile everything

**Comment by wftxd** [comment]
Oh, I see. Thank you!
